### PR TITLE
Clarify documentation about customising extensions

### PR DESCRIPTION
**Pull request by gmaclennan**
I found the documentation about customizing extensions unclear, and missed the note about browserify options. The examples given do not actually work by default because they do not set browserify options. This makes it clearer, I think.

### CODE REVIEW

**Comment by zertosh** [comment]
I like it for the most part, but it's a bit wordy and I think the example should cover the case of an extension babel already recognizes. What do you think of this?

By default, browserify only looks up `.js` and `.json` files when no extension is specified. To look up additional extensions, use browserify's [extensions](https://github.com/substack/node-browserify#browserifyfiles--opts) option. 

For example, so `require('./script')` is resolved to `./script.babel`:

[[your example]]

But, so `require('./component')` is resolved to `./component.jsx`:

```
browserify({ extensions: [".jsx"] }).transform(babelify)
```

```
$ browserify -d -e script.js --extensions=.jsx -t [ babelify ]
```

**Comment by MatthewHerbst** [comment]
@zertosh is it possible for Babelify to automatically add the '.jsx' extension to Browserify when the transform is applied? I just ran into this issue, and I feel that if Babelify is going to have support for JSX on by default, then it should also support '.jsx' component files by default.

**Comment by zertosh** [comment]
@MatthewHerbst sorry I don't think it's prudent to push that as a default. That's changing the semantics of the `require` function, so I think it's best left as an opt-in. Also, supporting JSX syntax isn't necessarily tied to ".jsx" files. For example, another common convention is to use ".react.js".

**Comment by MatthewHerbst** [comment]
@zertosh sure, I see what you're saying. What it be possible to maybe add a note about file extensions (maybe using JSX as an example) in the [description of the `nonStandard` option](https://babeljs.io/docs/usage/options/)?

For example, changing

> Enable support for JSX and Flow.

to something such as

> Enable support for JSX and Flow. (To enable support for non-standard file extensions such as .jsx, see [Customizing extensions])

**Comment by zertosh** [comment]
@MatthewHerbst I was hoping that this PR would've incorporated my edits. If you make a separate PR that clarifies the docs, I'd be more than happy to merge it :smile:

**Comment by MatthewHerbst** [comment]
@zertosh https://github.com/babel/babel.github.io/pull/430 :)

**Comment by zertosh** [comment]
The docs just got a lot clearer on this. Closing.
